Makefile

```
@@ -252,7 +252,7 @@ E2E_REGISTRY_IMAGE=localhost/e2e-test-registry:devel
 image-registry: export GOOS=linux
 image-registry: export GOARCH=amd64
 image-registry: ## Build the testdata catalog used for e2e tests and push it to the image registry
-	go build $(GO_BUILD_FLAGS) $(GO_BUILD_EXTRA_FLAGS) -tags '$(GO_BUILD_TAGS)' -ldflags '$(GO_BUILD_LDFLAGS)' -gcflags '$(GO_BUILD_GCFLAGS)' -asmflags '$(GO_BUILD_ASMFLAGS)' -o ./testdata/push/bin/push         ./testdata/push/push.go
+	go build $(GO_BUILD_FLAGS) $(GO_BUILD_EXTRA_FLAGS) -tags '$(GO_BUILD_TAGS)' -ldflags "$(GO_BUILD_LDFLAGS)" -gcflags '$(GO_BUILD_GCFLAGS)' -asmflags '$(GO_BUILD_ASMFLAGS)' -o ./testdata/push/bin/push         ./testdata/push/push.go
```

```suggestion
	go build $(GO_BUILD_FLAGS) $(GO_BUILD_EXTRA_FLAGS) \
    -tags '$(GO_BUILD_TAGS)' \
    # Use double quotes (not single quotes) for build flags like -ldflags, -tags, etc.
    # Single quotes are passed literally and not stripped by `go build`, which may cause errors
    # or inject unwanted characters into the binary (e.g., version strings).
    -ldflags "$(GO_BUILD_LDFLAGS)" \ 
    -gcflags '$(GO_BUILD_GCFLAGS)' \
    -asmflags '$(GO_BUILD_ASMFLAGS)' \
    -o ./testdata/push/bin/push \
    ./testdata/push/push.go
```

I think is better we format in a way that make it clear and to avoid mistakes we clarify here why just this one is using double quotes

---

Sure we can have the comment here too, can't hurt. Changing in next commit